src/tests/ftest/util/server_utils.py

```diff
@@ -50,320 +44,275 @@ class ServerFailed(Exception):
+
+    @property
+    def hosts(self):
+        """Hosts attribute getter."""
+        return self._hosts
+
+    @hosts.setter
+    def hosts(self, value):
+        """Hosts attribute setter
+
+        Args:
+            value (list): list of hosts
+        """
+        self._hosts = value
+        self.runner.processes.value = len(self._hosts)
+
+    @property
+    def hostfile(self):
+        """Hostfile attribute setter."""
+        return self._hostfile
+
+    @hostfile.setter
+    def hostfile(self, value):
+        """Hostfile attribute setter.
+
+        Args:
+            value (str): path to hostfile.
+        """
+        self._hostfile = value
+        self._hosts = get_hosts_from_file(value)
+        self.runner.hostfile.value = value
+        self.runner.processes.value = len(self._hosts)
+
+    def set_server_yaml(self):
+        """Set the server's config option to the path of the yaml file.
+
+        Args:
+            basepath (str): DAOS install basepath
+            log_filename (str): Set a specific logfile name
+        """
+        self.server_command.config.value = self._create_yaml(
+            self.basepath, self.log_filename)
+
+    def prep(self):
+        """Prepare the hosts before starting daos server.
+
+        Args:
+            hosts (list): list of host names to prepare
+
+        Raises:
+            ServerFailed: if there is any errors preparing the hosts
+
+        """
+        # Kill any doas servers running on the hosts
+        self.kill(self.hosts)
+
+        # Clean up any files that exist on the hosts
+        self.clean(self.hosts, scm=True)
+
+        # Verify the domain socket directory is present and owned by this user
+        user = getpass.getuser()
+        file_checks = (
+            ("Server", self.hosts, "/var/run/daos_server"),
+        )
+        for host_type, host_list, directory in file_checks:
+            status, nodeset = check_file_exists(host_list, directory, user)
+            if not status:
+                raise ServerFailed(
+                    "{}: {} missing directory {} for user {}.".format(
+                        nodeset, host_type, directory, user))
+
+    def start(self, mode="normal"):
+        """Start the server in normal mode or maintanence mode.
+
+        Args:
+            mode (str, optional): If "normal", will start server and wait
+                for IO servers to be started. If maintanence mode selected,
+                will wait for server to ask for storage format and return
+                when it does. Defaults to "normal".
+        """
+        patterns = {
+            "mtnc": "needs format (mountpoint doesn't exist)",
+            "normal": "DAOS I/O server.*started",
+        }
+        # Run servers
+        result = self.runner.run()
+
+        # Check for pattern
+        if mode in patterns:
+            pattern = patterns[mode]
+            poll_pattern(len(self.hosts), result, pattern)
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'pattern' in function call